Sources/NIOSSL/SSLContext.swift

```diff
@@ -508,15 +508,23 @@ extension NIOSSLContext {
             // This could be from a location like /etc/ssl/cert.pem as an example.
             CNIOBoringSSL_SSL_CTX_set_client_CA_list(context, CNIOBoringSSL_SSL_load_client_CA_file(path))
         } else if sendCANames, isDirectory {
-            // If the path that is passed in is a directory, scan the directory and gather up the PEM or DER files.
-            let pemFilePaths = DirectoryContents(path: path).filter { $0.suffix(4) == ".pem" || $0.suffix(4) == ".cer" }
-            // Create the PEM files one by one and use `addCACertificateNameToList` to add the CA name to the STACK_OF(X509_NAME).
-            for path in pemFilePaths {
+            // Match the c_rehash directory format and load the certificate based on this criteria.
+            let certificateFilePaths = try DirectoryContents(path: path).filter {
+                try self._isRehashFormat(path: $0)
+            }
+            // Load only the certificates that resolve to an existing certificate in the directory.
+            for symPath in certificateFilePaths {
+                guard let resolvedPath = try self.resolveSymlink(path: symPath) else { continue }
```

Comment:
Sorry I didn't spot this earlier, but I actually don't think we need this. We'll follow the symlink when we try to open it. Additionally, I believe c_rehash only supports PEM files, not DER, so I don't think that should be an issue either.

Reply:
@Lukasa you are correct about the DER support for `c_rehash`.  I just ran another test with a `PEM` and a `CER` file and only seen one hash output in the directory.  I guess I was confused by the docs for [c_rehash](https://www.openssl.org/docs/man1.0.2/man1/c_rehash.html) here:

> c_rehash scans directories and calculates a hash value of each .pem, .crt, .cer, or .crl file in the specified directory list and creates symbolic links for each file, where the name of the link is the hash value.

I will refactor this.